controllers/volumereplicationgroup_controller.go

```diff
@@ -705,16 +705,41 @@ func (v *VRGInstance) deleteVRGHandleMode() bool {
 	return v.reconcileVRsForDeletion()
 }
 
+// addFinalizer adds a finalizer to VRG, to act as deletion protection
+func (v *VRGInstance) addFinalizer(finalizer string) error {
```

Comment:
@hatfieldbrian you moved this function here and refactored it a little bit, that's good.
But what's the change besides that? I am unable to tell.

Reply:
@BenamarMk Lines 715 and 724 save and restore status just like `removeFinalizer` below

Reply:
In my experience it does update the local copy of the object.  A comment in the operator-sdk [documentation](https://sdk.operatorframework.io/docs/building-operators/golang/references/client/#update) suggests the same:
> // Update updates the given obj in the Kubernetes cluster. obj must be a
// struct pointer so that obj can be updated with the content returned
// by the API server. Update does *not* update the resource's status
// subresource
func (c Client) Update(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error

I presume the part about not updating the status refers to the API server copy of the object only.

Reply:
Ok, I didn't know that the original object gets updated. Interesting...